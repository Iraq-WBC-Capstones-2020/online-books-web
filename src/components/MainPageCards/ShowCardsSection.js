import React, { useState, useEffect } from 'react';
import './style.scss';
import MainCard from './MainCard';
import { MDBRow, MDBBtn, MDBIcon } from 'mdbreact';
import '../CarsouelSlick/carsouelSlick.scss';
import 'slick-carousel/slick/slick.css';
import 'slick-carousel/slick/slick-theme.css';
import Slider from 'react-slick';

function ShowCardsSection({ books }) {
  const useWindowSize = () => {
    const [size, setSize] = useState(window.innerWidth);
    useEffect(() => {
      const handleResize = () => {
        setSize([window.innerWidth]);
      };
      window.addEventListener('resize', handleResize);
    }, []);
    return size;
  };
  const width = useWindowSize();
  const isSmallScreen = width <= 425 ? true : false;

  const NextArrow = ({ onClick }) => {
    return (
      <div className="absolute rightArrowOfGrid">
        <MDBBtn color="" className="shadow-md p-0 h-8 " onClick={onClick}>
          <MDBIcon icon="angle-double-right" size="lg" className="px-2 py-1" />
        </MDBBtn>
      </div>
    );
  };

  const PrevArrow = ({ onClick }) => {
    return (
      <div className="absolute z-10 leftArrowOfGrid">
        <MDBBtn color="" className="shadow-md p-0 h-8" onClick={onClick}>
          <MDBIcon icon="angle-double-left" size="lg" className="px-2 py-1" />
        </MDBBtn>
      </div>
    );
  };

  const settings = {
    dots: true,
    dotsClass: 'slick-dots slick-thumb dotsMargin',
    infinite: true,
    slidesToShow: isSmallScreen ? 2 : 4,
    speed: 500,
    rows: 1,
    slidesPerRow: 2,
    nextArrow: <NextArrow />,
    prevArrow: <PrevArrow />,
  };
  return (
    <>
      {/*flex-col justify-between className="relative"*/}
      <MDBRow className="containerGridCards">
        <Slider {...settings} className="w-full relative">
          {books.map((book) => (
            <MainCard book={book} key={book.id} />
          ))}
        </Slider>
      </MDBRow>
    </>
  );
}

export default ShowCardsSection;
