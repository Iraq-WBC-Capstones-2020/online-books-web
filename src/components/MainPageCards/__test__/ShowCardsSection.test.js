import React from 'react';
import renderer from 'react-test-renderer';
import ShowCardsSection from './../ShowCardsSection';

test('ShowCardsSection snapshot', () => {
  const tree = renderer.create(<ShowCardsSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
