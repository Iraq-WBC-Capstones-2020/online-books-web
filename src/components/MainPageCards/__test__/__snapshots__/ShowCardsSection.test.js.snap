// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowCardsSection snapshot 1`] = `
  <>
      <div className="containerGridCards flex-col justify-between" data-test="row">
        {BookInfo.map((book) => (
          <MainCard book={book} key={book.id} />
        ))}
      </div>
    </>
`;
